lib/renderer/react.js

```diff
@@ -23,76 +24,95 @@ reactExport.configureReact = async (templateLocation, templateContentDir, transp
  * @param {boolean} debug flag
- * @param {string} originalAsyncAPI 
+ * @param {string} originalAsyncAPI
  * @return {Promise<TemplateRenderResult>}
  */
 reactExport.renderReact = async (asyncapiDocument, filePath, extraTemplateData, templateLocation, templateContentDir, transpiledTemplateLocation, templateParams, debug, originalAsyncAPI) => {
   extraTemplateData = extraTemplateData || {};
   filePath = filePath.replace(templateContentDir, path.resolve(templateLocation, transpiledTemplateLocation));
   return await AsyncReactSDK.renderTemplate(
-    filePath, 
+    filePath,
     {
       asyncapi: asyncapiDocument,
       params: templateParams,
       originalAsyncAPI,
       ...extraTemplateData
-    }, 
+    },
     debug
   );
 };
 
 /**
  * Save the single rendered react content based on the meta data available.
- * 
+ *
  * @private
  * @param {TemplateRenderResult} renderedContent the react content rendered
  * @param {String} outputPath Path to the file being rendered.
+ * @param {String[]} noOverwriteGlobs globs to check for files that should not be overwritten.
  */
-const saveContentToFile = async (renderedContent, outputPath) => {
+const saveContentToFile = async (renderedContent, outputPath, noOverwriteGlobs = []) => {
   let filePath = outputPath;
-  // Might be the same as in the `fs` package, but is an active choice for our default file permission for any rendered files. 
+  // Might be the same as in the `fs` package, but is an active choice for our default file permission for any rendered files.
   let permissions = 0o666;
   const content = renderedContent.content;
 
+  // Error handling for undefined content or non-string type
   if (content === undefined || typeof content !== 'string') {
     throw Error(`Template file not rendered correctly, was rendered as ${content}`);
   }
+
+  // Adjust file permissions based on metadata, if available
   if (renderedContent.metadata !== undefined) {
     if (renderedContent.metadata.permissions !== undefined) {
       permissions = renderedContent.metadata.permissions;
     }
+
+    // Replace the base filename with a new filename from metadata if specified
     if (renderedContent.metadata.fileName !== undefined) {
       const newFileName = renderedContent.metadata.fileName.trim();
       const basepath = path.basename(filePath);
       filePath = filePath.replace(basepath, newFileName);
     }
   }
 
-  await writeFile(filePath, content, {
-    mode: permissions
-  });
+  // get the final file name of the file
+  const finalFileName = path.basename(filePath);
+  // check whether the filename should be ignored based on user's inputs
+  const shouldOverwrite = !noOverwriteGlobs.some(globExp => minimatch(finalFileName, globExp));
+
+  // Write the file only if it should not be skipped
+  if (shouldOverwrite) {
+    await writeFile(filePath, content, {
+      mode: permissions
+    });
+  } else {
+    console.log(`Skipping overwrite for: ${filePath}`);
```

Comment:
all is good except of this one. Such logs should only be visible on demand, with debug flag

have a look into `generator.js` as the usage of logger, here you also should log out info with `log.debug`

Reply:
I just updated it! I think it should be appropriate to use `const log = require('loglevel');` here, since it's loaded in the generator into the cache already.